### PR TITLE
Usage of gtest_discover_tests [19810]

### PR DESCRIPTION
**Pull request by richiware**
## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- *N/A* Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- *N/A* Any new/modified methods have been properly documented using Doxygen. 
- *N/A* Changes are ABI compatible. 
- *N/A* Changes are API compatible. 
- *N/A* New feature has been added to the `versions.md` file (if applicable).
- *N/A* New feature has been documented/Current behavior is correctly described in the documentation. 
    
- *N/A* Applicable backports have been included in the description.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Comment by richiware** [comment]
@richiprosima please test this

**Comment by richiware** [comment]
@richiprosima please test this

**Comment by richiware** [comment]
@richiprosima please test mac

**Comment by richiware** [comment]
@richiprosima please test this

**Comment by richiware** [comment]
@richiprosima please test this

**Comment by adriancampo** [comment]
@richiprosima please test this

File: test/unittest/transport/SharedMemTests.cpp [hunk 1]
```diff
@@ -181,12 +181,14 @@ class SHMRingBuffer : public ::testing::Test
...
+/*
+   class SHMRingBufferMultiThread
     :   public SHMRingBuffer,
     public testing::WithParamInterface<std::tuple<uint32_t, uint32_t, uint32_t>>
-{
+   {
 
-};
+   };
+ */
```
Comment (adriancampo) [inline test/unittest/transport/SharedMemTests.cpp:191]: This should follow whatever happends with the `SHMRingBufferMultiThread` test.

File: test/unittest/transport/SharedMemTests.cpp [hunk 2]
```diff
@@ -364,8 +366,9 @@ TEST_F(SHMRingBuffer, listeners_register_unregister)
     listener2->pop();
 }
 
-TEST_P(SHMRingBufferMultiThread, multiple_writers_listeners)
-{
+/*
+   TEST_P(SHMRingBufferMultiThread, multiple_writers_listeners)
```
Comment (adriancampo) [inline test/unittest/transport/SharedMemTests.cpp:370]: Should this test be left commented, fixed, or removed?

Comment (richiware) [reply test/unittest/transport/SharedMemTests.cpp]: Removed in c11f830d7

**Comment by richiware** [comment]
@richiprosima please test this

**Review by adriancampo — DISMISSED** [review]
LGTM

**Comment by richiware** [comment]
@richiprosima please test this

**Review by adriancampo — DISMISSED** [review]

**Comment by richiware** [comment]
@richiprosima please test mac

**Comment by richiware** [comment]
@richiprosima please test mac

File: test/unittest/transport/SharedMemTests.cpp [hunk 3]
```diff
@@ -364,110 +357,6 @@ TEST_F(SHMRingBuffer, listeners_register_unregister)
     listener2->pop();
 }
 
-TEST_P(SHMRingBufferMultiThread, multiple_writers_listeners)
```
Comment (MiguelCompany) [inline test/unittest/transport/SharedMemTests.cpp:367]: Why is this test being removed?

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

**Comment by richiware** [comment]
@richiprosima please test this

File: test/unittest/dds/topic/DDSSQLFilter/CMakeLists.txt [hunk 4]
```diff
@@ -50,14 +50,4 @@ target_include_directories(DDSSQLFilterTests PRIVATE
...
-# Tests with specific timeout
-if (GTEST_INDIVIDUAL)
-    # This test does not exist unless it is compiled with GTEST_INDIVIDUAL
-    set_tests_properties("DDSSQLFilterValueTests.test_filtered_value" PROPERTIES TIMEOUT 500)
-endif ()
-
-if(ANDROID)
-    set_property(TARGET DDSSQLFilterTests PROPERTY CROSSCOMPILING_EMULATOR "adb;shell;cd;${CMAKE_CURRENT_BINARY_DIR};&&")
-endif()
+gtest_discover_tests(DDSSQLFilterTests PROPERTIES TIMEOUT 500)
```
[the remark this answers is not in the document]
Comment (richiware) [reply test/unittest/dds/topic/DDSSQLFilter/CMakeLists.txt]: Done in 5826c5f

File: test/unittest/transport/SharedMemTests.cpp (hunk 3 above)
Comment (richiware) [reply test/unittest/transport/SharedMemTests.cpp:367]: We decided to remove it. It is commented from the beginning and never works

**Review by MiguelCompany — APPROVED** [review]

**Comment by richiware** [comment]
@richiprosima  please test mac

**Comment by richiware** [comment]
@richiprosima please test windows

**Comment by richiware** [comment]
@richiprosima please test mac

**Pull request merged**